### PR TITLE
M3-4753: Convert ThemeToggle to function component & update ThemeToggle.test.tsx

### PR DESCRIPTION
**Pull request by DevDW**
## Description
Follow-up to https://github.com/linode/manager/pull/7242.

Converted `ThemeToggle.tsx` to a function component, updated `ThemeToggle.test.tsx` and moved it to the Profile/Settings folder, and made some minimal touches to a few `PrimaryNav` files for TypeScript purposes

## Type of Change
- Non breaking change ('update', 'change')

### CODE REVIEW

File: packages/manager/src/components/PrimaryNav/index.ts [hunk 1]
```diff
@@ -1,2 +1,2 @@
-import PrimaryNav from './PrimaryNav';
+import PrimaryNav from './PrimaryNav_CMR';
```
Comment (Jskobos) [inline packages/manager/src/components/PrimaryNav/index.ts:1]: Should probably rename PrimaryNav_CMR to PrimaryNav now that the non-CMR file is gone

File: packages/manager/src/features/Profile/Settings/ThemeToggle.tsx [hunk 2]
```diff
@@ -19,31 +13,29 @@ const onClickHandler = () => {
...
+          [themeName]: true
+        })}
+        aria-label="Switch Theme"
+      />
+    </div>
+  );
+};
 
-const styled = withStyles(styles, { withTheme: true });
+const styled = withStyles({}, { withTheme: true });
```
Comment (Jskobos) [inline packages/manager/src/features/Profile/Settings/ThemeToggle.tsx:39]: This isn't necessary, there's a `useTheme` hook you can use.

File: packages/manager/src/components/PrimaryNav/index.ts (hunk 1 above)
Comment (Jskobos) [reply packages/manager/src/components/PrimaryNav/index.ts:1]: And there's a PrimaryNav.styles.ts and CMR version that should be deleted/moved too.

**Review by tiffwong — APPROVED** [review]
Other than cleanup items already mentioned, everything else looks good!

**Review by Jskobos — APPROVED** [review]

File: packages/manager/src/features/Profile/Settings/ThemeToggle.test.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { shallow } from 'enzyme';
```
Comment (Jskobos) [inline packages/manager/src/features/Profile/Settings/ThemeToggle.test.tsx:1]: This is failing still. Might as well swap in RTL and use renderWithTheme which hopefully will take care of it.

**Review by johnwcallahan — APPROVED** [review]

**Pull request merged**